website/docs/advanced-topics/struct-components/context-consumption.mdx

```diff
@@ -0,0 +1,77 @@
+---
```

Comment:
This seems more like an example embedded in a doc page than documentation. We run into the same issue as #2189

Reply:
Yeah but i had to answer this so many times on discord that having a page about it wont hurt

Reply:
Also adding a full package in `examples/` seems a bit overkill imo
Maybe introduce it in some existing example?
But in that case I would still prefer having a page i can quickly link to and not have to go to github and click through bunch of folders.

Reply:
We could add a struct counterpart to this component: https://github.com/yewstack/yew/blob/master/examples/contexts/src/subscriber.rs